prov/rxm/src/rxm_conn.c

```diff
@@ -624,26 +624,37 @@ static void *rxm_conn_eq_read(void *arg)
 static void *rxm_conn_progress(void *arg)
 {
 	struct rxm_ep *rxm_ep = container_of(arg, struct rxm_ep, util_ep);
-	struct rxm_msg_eq_entry entry;
+	struct rxm_msg_eq_entry *entry;
 	int ret;
 
+	entry = calloc(1, RXM_MSG_EQ_ENTRY_SZ);
+	if (!entry) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+			"Unable to allocate memory!\n");
+		return NULL;
+	}
 	FI_DBG(&rxm_prov, FI_LOG_EP_CTRL, "Starting conn event handler\n");
+
 	while (1) {
-		memset(&entry, 0, sizeof(entry));
-		entry.rd = rxm_eq_sread(rxm_ep, RXM_CM_ENTRY_SZ, &entry);
-		if (entry.rd < 0 && entry.rd != -FI_ECONNREFUSED)
-			return NULL;
+		entry->rd = rxm_eq_sread(rxm_ep, RXM_CM_ENTRY_SZ, entry);
```

Comment:
It's confusing seeing calloc() for RXM_MSG_EQ_ENTRY_SZ, but passing in RXM_CM_ENTRY_SZ here.  Can we use RXM_MSG_EQ_ENTRY_SZ in both places?

Reply:
It is somewhat confusing at first glance, but it is required based on how rxm_eq_sread is written. The fi_eq_cm_entry structure is embedded in the rxm_msg_eq_entry structure, but the length parameter (RXM_CM_ENTRY_SZ) passed specifies only the maximum size of the embedded CM entry and any data.